XCUITests/HomePageSettingsUITest.swift

```diff
@@ -167,4 +168,57 @@ class HomePageSettingsUITests: BaseTestCase {
         // There is one entry
         XCTAssertEqual(app.tables["History List"].cells.count, 3)
     }
+    
+    func testTopSitesCustomNumberOfRows() {
+        //Pre-populate top sites with 16 URLs so we can see all 4 rows
+        for i in 0..<14 {
+            pinEnoughSitesToFillFourRows(urlStr: urlList[i])
+        }
+        navigator.goto(HomeSettings)
+        //Check whether or not each of the custom row options changes UI and works correctly
+        changeNumberofRows(numRows: "1")
+        XCTAssertEqual(app.tables.cells["TopSitesRows"].label as String, "Top Sites, Rows: 1")
+        navigator.performAction(Action.GoToHomePage)
+        checkNumberOfExpectedTopSites(numberOfExpectedTopSites: 4)
+        changeNumberofRows(numRows: "2")
+        XCTAssertEqual(app.tables.cells["TopSitesRows"].label as String, "Top Sites, Rows: 2")
+        navigator.performAction(Action.GoToHomePage)
+        checkNumberOfExpectedTopSites(numberOfExpectedTopSites: 8)
+        changeNumberofRows(numRows: "3")
+        XCTAssertEqual(app.tables.cells["TopSitesRows"].label as String, "Top Sites, Rows: 3")
+        navigator.performAction(Action.GoToHomePage)
+        checkNumberOfExpectedTopSites(numberOfExpectedTopSites: 12)
+        changeNumberofRows(numRows: "4")
+        XCTAssertEqual(app.tables.cells["TopSitesRows"].label as String, "Top Sites, Rows: 4")
+        navigator.performAction(Action.GoToHomePage)
+        checkNumberOfExpectedTopSites(numberOfExpectedTopSites: 16)
+    }
+    //Function to check the number of top sites shown given a selected number of rows
+    private func checkNumberOfExpectedTopSites(numberOfExpectedTopSites: Int) {
+        waitForExistence(app.cells["TopSitesCell"])
+        XCTAssertTrue(app.cells["TopSitesCell"].exists)
+        let numberOfTopSites = XCUIApplication().collectionViews.cells["TopSitesCell"].cells.matching(identifier: "TopSite").count
+        XCTAssertEqual(numberOfTopSites, numberOfExpectedTopSites)
+    }
+    //Function to pre-populate the number of pinned sites with enough sites to show 4 rows
+    private func pinEnoughSitesToFillFourRows(urlStr: String) {
+        navigator.openURL(urlStr)
+        navigator.goto(PageOptionsMenu)
+        navigator.performAction(Action.PinToTopSitesPAM)
+    }
+    //Function to contain the commands to change the number of rows of top sites
+    private func changeNumberofRows(numRows: String) {
+        navigator.goto(HomeSettings)
+        waitForExistence(app.tables.cells["TopSitesRows"])
+        app.tables.cells["TopSitesRows"].tap()
+        waitForExistence(app.staticTexts["SET ROWS"])
+        XCTAssertTrue(app.staticTexts["SET ROWS"].exists)
+        XCTAssertTrue(app.staticTexts["1"].exists)
+        XCTAssertTrue(app.staticTexts["2"].exists)
+        XCTAssertTrue(app.staticTexts["3"].exists)
+        XCTAssertTrue(app.staticTexts["4"].exists)
+        app.staticTexts[numRows].tap()
```

Comment:
Not sure if we could add an action for this in the [HomeSettings Screen ](https://github.com/mozilla-mobile/firefox-ios/blob/master/XCUITests/FxScreenGraph.swift#L652) Also next line could be included as part of the back action and so the test would be more readable...otherwise is difficult to know that that line (below line) does